src/cleanfigure.m

```diff
@@ -799,30 +845,87 @@ function simplifyLine(meta, handle, targetResolution)
   if isYlog
     yLim  = log10(yLim);
   end
+  if is3D
+      isZlog = strcmp(get(meta.gca, 'ZScale'), 'log');
+      if isZlog
+        zLim  = log10(zLim);
+      end
+  end
+
+  % In case of 3D plots, project the limits into the image plane. Depending
+  % on the angles, any of the 8 corners of the 3D cube mit be relevant so
+  % check for all
+  if is3D
+	% Get the projection angle
+    [az, el] = view(meta.gca);
+
+    % Convert from degrees to radians.
+    az = az*pi/180;
+    el = el*pi/180;
+
+    % The transformation into the image plane is done i a two-step process.
+    % First rotate around the z-axis by -az (in radians)
+    % Second rotate around the x-axis by (el -pi/2) radians.
```

Comment:
Nice to see a lot of comments, however, I would place the `Second, rotate ...` more towards the bottom. You can even make that into a cell comment (i.e. ``%% Rotate around x-axis over (el - pi/2) radians`)